lib/galaxy/datatypes/text.py

```diff
@@ -252,10 +253,24 @@ def _transform_dict_list_ids(dict_list):
                                          ('table_columns', 'columns')]:
                     try:
                         metadata_value = json_dict.get(b_name, None)
+                        if b_name == "columns" and metadata_value:
+                            mks = metadata_value[0]['metadata'].keys()
+                            keep_columns = { mk: None for mk in mks }
```

Comment:
Can you remove the extra spaces to make flake8 happy? https://travis-ci.org/galaxyproject/galaxy/jobs/451325707

```suggestion
                            keep_columns = {mk: None for mk in mks}
```

Reply:
Done! Cheers again @nsoranzo